File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
int main(void)
{
    string text = get_string("TEXT: ");
    int len = strlen(text);
    int wordCounter = 0;
    int latterCounter =0;
    int sentenceCounter =1;
    int i;
    if(len >= 0)
    {
            for (i= 0; i<len-1;i++ )
        {
            if(text[i]==' ')
            {
                if (i != 0 && text[i-1] != ' ')
                {
                    wordCounter ++;
                }

            }
            if( ((int)text[i] >= 65 && (int)text[i] <= 90) || ((int)text[i] >= 97 && (int)text[i] <= 122) )
            {
               latterCounter ++;
            }
            if (  text[i] == '.' || text[i] == '!' || text[i] == '?')
            {
                sentenceCounter ++;
            }

        }
        wordCounter++;
    }
    float avarageLet;
    float avarageSen;
    avarageLet = ((float)latterCounter / (float)wordCounter)*100;
    avarageSen = ((float)sentenceCounter / (float)wordCounter) * 100;
    int Grade =  (int) round((0.0588*avarageLet) - (0.296 * avarageSen) -15.8);
    if(Grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if(Grade < 1 )
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", Grade);
    }

}

File: readability.py
import cs50

text = cs50.get_string("Text :")
words = text.count(" ") + 1
latters = len(''.join(i for i in text if i.isalpha()))
sentance = text.count(".")+ text.count("!")+text.count("?")
grade = (0.0588 * (latters / words * 100)) - (0.296 * (sentance / words) * 100) - 15.8
if grade < 1 :
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(round(grade)))